### PR TITLE
Alert: Add dark theme

### PR DESCRIPTION
**Pull request by island205**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: packages/theme-chalk/src/alert.scss [hunk 1]
```diff
@@ -15,6 +15,15 @@
   align-items: center;
   transition: opacity .2s;
 
+  &.is-darken {
+    .el-alert__closebtn {
```
Comment (iamkun) [inline packages/theme-chalk/src/alert.scss]: 移到 `@include e(closebtn) {` 更好维护吧

Comment (island205) [reply packages/theme-chalk/src/alert.scss]: 讨论下来，这个问题不存在了。

Comment (iamkun) [reply packages/theme-chalk/src/alert.scss]: 就当无事发生 😬

File: examples/docs/es/alert.md [hunk 2]
```diff
@@ -30,7 +30,38 @@ Los componentes de alertas no son elementos overlay de la página y no desaparec
...
+
+Match the main colors which are darker.
+
+:::demo Alert provide `darken` property to enable the dark theme.
+```html
+<template>
+  <el-alert
+    title="success alert"
+    type="success"
+    darken>
```
Comment (wacky6) [inline examples/docs/es/alert.md]: 直接用 dark 就好了。 darken 是动词，当作 attr 有点怪。 

css里面 is-darken / is-darkened 就更奇怪了 😂 直接 is-dark 吧

Comment (island205) [reply examples/docs/es/alert.md]: 改了

**Comment by ziyoung** [comment]
添加新属性：effect，默认值为 light。

File: packages/theme-chalk/src/alert.scss [hunk 3]
```diff
@@ -15,43 +15,86 @@
...
+    &.is-dark {
+      background-color: $--color-success;
+      color: $--color-white;
     }
   }
 
   @include m(info) {
-    background-color: $--alert-info-color;
-    color: $--color-info;
+    &.is-light {
```
Comment (ziyoung) [inline packages/theme-chalk/src/alert.scss]: 可以这样写：
```scss
@include when(light) {
// ...
}
```

File: types/alert.d.ts [hunk 4]
```diff
@@ -24,4 +24,7 @@ export declare class ElAlert extends ElementUIComponent {
 
   /** If a type icon is displayed */
   showIcon: boolean
+
+  /** Choose theme */
+  theme: string
```
Comment (ziyoung) [inline types/alert.d.ts]: 再声明一个类型吧。
```ts
export type AlertEffect = 'dark' | 'light'
// ...
theme: AlertEffect
```

**Pull request merged**